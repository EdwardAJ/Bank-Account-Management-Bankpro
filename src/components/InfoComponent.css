.info {
    color: white;
    line-height: 0.5rem;
}

.info p { 
    font-size: 0.85rem;
}

.info h3 {
    font-size: 1.2rem;
    margin-bottom: 2rem;
}