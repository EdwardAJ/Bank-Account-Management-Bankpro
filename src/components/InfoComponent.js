import React from 'react';
import './InfoComponent.css';

import axios from 'axios';
import { INFO_URL } from './../utils/endpoints.js';
import { HEADERS } from './../utils/header.js';
import getXMLResponse from './../utils/functions.js';
import { getCookie } from './../utils/cookie.js';

export default class InfoComponent extends React.Component {
    constructor (props) {
        super(props);
        this.state = {
            nama: '',
            noRek: '',
            bank: '',
            saldo: ''
        };
        this.fetchUserInfo();
    }

    fetchUserInfo() {
        let requestBody =
        '<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:con="http://controllers.webservice.com/">' +
            '<soapenv:Header/>' +
            '<soapenv:Body>' +
                '<con:fetchUserInfo>' +
                    '<accountNumber>' + getCookie("auth") + '</accountNumber>' +
                '</con:fetchUserInfo>' +
            '</soapenv:Body>' +
        '</soapenv:Envelope>';

        axios.post(INFO_URL, requestBody, {
            headers: HEADERS
        }).then((response) => {
            if (response.status === 200) {
                let nama = getXMLResponse(response.data)[0].innerHTML;
                let noRek = getXMLResponse(response.data)[1].innerHTML;
                let bank = getXMLResponse(response.data)[2].innerHTML;
                let saldo = getXMLResponse(response.data)[3].innerHTML;
                this.setState({ nama: nama, noRek: noRek, bank: bank, saldo: saldo });
            }
        }).catch((error) => {
            alert("Error: ", error);
        });
    }

    render() {
        let nama = this.state.nama;
        let noRek = this.state.noRek;
        let bank = this.state.bank;
        let saldo = this.state.saldo;
        return (
            <div className="info">
                <p> Nasabah </p>
                <h3> { nama } </h3>
                <p> Nomor Rekening </p>
                <h3> { noRek } </h3>
                <p> Bank </p>
                <h3> { bank } </h3>
                <p> Saldo Terakhir </p>
                <h3> { saldo } </h3>  
            </div>
        );
    }
}