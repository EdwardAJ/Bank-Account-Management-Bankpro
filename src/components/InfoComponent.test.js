import 'jsdom-global/register';
import React from 'react';
import { shallow } from 'enzyme';
import InfoComponent from './InfoComponent';

describe("InfoComponent", () => {
    it("should match this component", () => {
        const component = shallow(<InfoComponent />);
        expect(component.getElements()).toMatchSnapshot();
    });
});

describe("InfoComponent", () => {
    it("should render this component correctly", () => {
        const wrapper = shallow(<InfoComponent />);
        expect(wrapper.find('h3')).toHaveLength(4);
        expect(wrapper.find('p')).toHaveLength(4);
    });
});