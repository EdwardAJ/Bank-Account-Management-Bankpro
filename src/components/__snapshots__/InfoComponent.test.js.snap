exports[`AuthPage should match this component 1`] = `
Array [
  <div
    className="info">
    <p>
         Nasabah 
    </p>
    <h3>
         
        
         
    </h3>
    <p>
         Nomor Rekening 
    </p>
    <h3>
         
        
         
    </h3>
    <p>
         Bank 
    </p>
    <h3>
         
        
         
    </h3>
    <p>
         Saldo Terakhir 
    </p>
    <h3>
         
        
         
    </h3>
</div>,
]
`;

exports[`InfoComponent should match this component 1`] = `
Array [
  <div
    className="info">
    <p>
         Nasabah 
    </p>
    <h3>
         
        
         
    </h3>
    <p>
         Nomor Rekening 
    </p>
    <h3>
         
        
         
    </h3>
    <p>
         Bank 
    </p>
    <h3>
         
        
         
    </h3>
    <p>
         Saldo Terakhir 
    </p>
    <h3>
         
        
         
    </h3>
</div>,
]
`;
